models/articles.model.js

```diff
@@ -0,0 +1,15 @@
+const db = require("../db/connection");
+
+exports.fetchArticleById = (article_id, next) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, message: "Not found" });
+      }
+      return rows[0];
+    })
+    .catch((err) => {
```

Comment:
We should only have the one .catch block in each promise chain, lets allow the one in the controller to handle any errors along this chain 👍